src/main/java/org/symphonyoss/symphony/messageml/elements/Code.java

```diff
@@ -30,29 +39,92 @@
 public class Code extends Element {
 
   public static final String MESSAGEML_TAG = "code";
+  private static final String MML_LANGUAGE_ATTR = "language";
+  private static final String PML_LANGUAGE_ATTR = "data-language";
   public static final char MARKDOWN_DELIMITER_CHAR = '`';
   public static final int MARKDOWN_DELIMITER_LENGTH = 3;
   private static final int MARKDOWN_DELIMITER_INDENT = 0;
 
+  private static final List<String> SUPPORTED_LANGUAGES = Arrays.asList(
+      "plaintext", "c", "cpp", "csharp", "css", "html", "java", "js", "jsx", "php", "python", "r", "typescript", "tsx"
+  );
+
   public Code(Element parent) {
     super(parent, MESSAGEML_TAG);
   }
+  public Code(Element parent, String language) {
+    super(parent, MESSAGEML_TAG);
+    this.setAttribute(PML_LANGUAGE_ATTR, language);
+  }
+
+  @Override
+  void buildAttribute(MessageMLParser parser, org.w3c.dom.Node item) throws InvalidInputException {
+    switch (item.getNodeName()) {
+      case MML_LANGUAGE_ATTR:
+      case PML_LANGUAGE_ATTR:
+        setAttribute(item.getNodeName(), getStringAttribute(item));
+        break;
+      default:
+        super.buildAttribute(parser, item);
+    }
+  }
+
+  @Override
+  void asPresentationML(XmlPrintStream out, MessageMLContext context) {
+    final Map<String, Object> attrs = new HashMap<>();
+
+    if (getAttribute(MML_LANGUAGE_ATTR) != null) {
+      attrs.put(PML_LANGUAGE_ATTR, getAttribute(MML_LANGUAGE_ATTR));
+    }
+
+    if (getAttribute(PML_LANGUAGE_ATTR) != null) {
+      attrs.put(PML_LANGUAGE_ATTR, getAttribute(PML_LANGUAGE_ATTR));
+    }
+
+    out.openElement(MESSAGEML_TAG, attrs);
+
+    for (Element child : getChildren()) {
+      child.asPresentationML(out, context);
+    }
+
+    out.closeElement();
+  }
 
   @Override
   public Node asMarkdown() {
-    FencedCodeBlock node = new FencedCodeBlock();
+    final FencedCodeBlock node = new FencedCodeBlock();
     node.setFenceChar(MARKDOWN_DELIMITER_CHAR);
     node.setFenceLength(MARKDOWN_DELIMITER_LENGTH);
     node.setFenceIndent(MARKDOWN_DELIMITER_INDENT);
+
+    if (getAttribute(MML_LANGUAGE_ATTR) != null) {
+      node.setInfo(getAttribute(MML_LANGUAGE_ATTR));
+    }
+
+    if (getAttribute(PML_LANGUAGE_ATTR) != null) {
+      node.setInfo(getAttribute(PML_LANGUAGE_ATTR));
+    }
+
     return node;
   }
 
   @Override
   public void validate() throws InvalidInputException {
-    assertNoAttributes();
```

Comment:
If the code has a different attribute than "language" and "data-language", is it still valid? Do we need to check that?

Reply:
<img width="1792" alt="Screen Shot 2022-02-14 at 19 28 51" src="https://user-images.githubusercontent.com/39826516/153924393-beb03c95-7084-4a45-866d-359da5c1554b.png">